### DISCUSSION TITLE
Sumatra PDF on Linux in wine (error with Mobi files)

### DISCUSSION THREAD

**Discussion by SpearRaven** [sumatrapdfreader/sumatrapdf#2880] [Q&A]
Hi,

I installed this in wine and can open PDF and EPUB files fine. But I get an error with Mobi files. Can you tell what dll file or whatever I need here?

Unhandled exception: page fault on read access to 0x0000000000000010 in 64-bit code (0x0000000140070c16).
Register dump:
 rip:0000000140070c16 rsp:000000000011e8e0 rbp:0000000140070c0c eflags:00010202 (  R- --  I   - - - )
 rax:0000000000000001 rbx:0000000000000000 rcx:00000000001f7720 rdx:0000000000000000
 rsi:000000000011e9f0 rdi:00000000001f7720  r8:0000000000000000  r9:0000000000140138 r10:0000000000250000
 r11:0000000000237478 r12:0000000000000000 r13:0000000000165b10 r14:00000000001f7670 r15:000000000011e9a0
Stack dump:
0x000000000011e8e0:  00000000001f7670 0000000000176440
0x000000000011e8f0:  000000000011e9a0 00000000001f7670
0x000000000011e900:  00000000001f7720 00000001400bf958
0x000000000011e910:  0000000000000000 0000000000000004
0x000000000011e920:  000000000011e9f0 0000000000000001
0x000000000011e930:  0000000000000000 0000000000000001
0x000000000011e940:  0000000000165b10 00000001400a94bb
0x000000000011e950:  0000000000000000 0000000000000000
0x000000000011e960:  0000000000183100 00000000001f7670
0x000000000011e970:  0000000000000000 00000001400ac4cf
0x000000000011e980:  0000000000000000 000000000011ea80
0x000000000011e990:  00000000001b5b70 00000000001b57c0
Backtrace:
=>0 0x0000000140070c16 in sumatrapdf (+0x70c16) (0x0000000140070c0c)
  1 0x00000001400bf958 in sumatrapdf (+0xbf958) (0x0000000140070c0c)
  2 0x00000001400ac4cf in sumatrapdf (+0xac4cf) (0x000000000011ea80)
  3 0x00000001400ac8cc in sumatrapdf (+0xac8cc) (0x0000000000000060)
  4 0x000000014009eb6b in sumatrapdf (+0x9eb6b) (0x0000000000000060)
  5 0x000000014009ed53 in sumatrapdf (+0x9ed53) (0x000000000011f2a0)
  6 0x000000014005362b in sumatrapdf (+0x5362b) (0x000000000011f2a0)
  7 0x00000001400551de in sumatrapdf (+0x551de) (0x000000000011f2e9)
  8 0x00000001400607ee in sumatrapdf (+0x607ee) (0x000000000011f450)
  9 0x0000000140062d1d in sumatrapdf (+0x62d1d) (0x000000000011f670)
  10 0x00000001400e985e in sumatrapdf (+0xe985e) (0x0000000000000000)
  11 0x000000007b62c719 BaseThreadInitThunk+0x9(unknown=<internal error>, entry=<internal error>, arg=<internal error>) [Z:\home\user\dlls\kernel32\thread.c:61] in kernel32 (0x0000000000000000)
  12 0x000000017005a263 __wine_pop_frame(unknown=<internal error>, entry=<internal error>, arg=<internal error>) [Z:\home\user\include\wine\exception.h:273] in ntdll (0x0000000000000000)
  13 0x000000017005a263 RtlUserThreadStart+0x83(entry=[<register RSP not accessible in this frame>, arg=[<register RSP not accessible in this frame>) [Z:\home\user\dlls\ntdll\thread.c:240] in ntdll (0x0000000000000000)
0x0000000140070c16 sumatrapdf+0x70c16: cmpq	$0,0x0000000000000010(%rdx)
Modules:
Module	Address					Debug info	Name (45 modules)
PE	000000007b000000-000000007b3f3000	Deferred        kernelbase
PE	000000007b600000-000000007b963000	Dwarf-4         kernel32
PE	0000000140000000-0000000140728000	Export          sumatrapdf
PE	0000000170000000-000000017031f000	Dwarf-4         ntdll
PE	0000000180000000-0000000180b32000	Deferred        libmupdf
PE	00000001c4ee0000-00000001c4fb4000	Deferred        winspool
PE	00000001c69e0000-00000001c776c000	Deferred        shell32
PE	00000001c8db0000-00000001c90c9000	Deferred        msvcrt
PE	00000001d97a0000-00000001d97ae000	Deferred        api-ms-win-core-fibers-l1-1-1
PE	00000001e7610000-00000001e761e000	Deferred        normaliz
PE	00000001ec2b0000-00000001ec350000	Deferred        ws2_32
PE	000000021a7e0000-000000021a9cc000	Deferred        setupapi
PE	0000000226310000-000000022631e000	Deferred        api-ms-win-core-string-l1-1-0
PE	0000000228450000-00000002284c6000	Deferred        propsys
PE	0000000231ae0000-0000000231d6c000	Deferred        rpcrt4
PE	000000023d820000-000000023e09c000	Deferred        user32
PE	0000000241b90000-0000000241bb9000	Deferred        zlib1
PE	0000000243540000-0000000243837000	Deferred        gdiplus
PE	000000024f470000-000000024f4d4000	Deferred        mpr
PE	000000025d740000-000000025d75f000	Deferred        dwmapi
PE	000000026b4c0000-000000026b756000	Deferred        gdi32
PE	00000002739c0000-0000000273ed1000	Deferred        oleaut32
PE	000000029ea80000-000000029ea8e000	Deferred        api-ms-win-core-datetime-l1-1-1
PE	00000002bb750000-00000002bbcd7000	Deferred        comctl32
PE	00000002e3540000-00000002e364f000	Deferred        shlwapi
PE	00000002e8f10000-00000002e949d000	Deferred        ole32
PE	00000002f1fa0000-00000002f1fbe000	Deferred        version
PE	00000002f7230000-00000002f72d6000	Deferred        uxtheme
PE	00000002fa030000-00000002fa03e000	Deferred        api-ms-win-core-localization-obsolete-l1-2-0
PE	000000030c980000-000000030c98f000	Deferred        api-ms-win-core-synch-l1-2-0
PE	00000003126f0000-0000000312745000	Deferred        shcore
PE	000000031f800000-000000031fa8e000	Deferred        comdlg32
PE	0000000327020000-00000003271ac000	Deferred        combase
PE	000000032a700000-000000032a7c1000	Deferred        sechost
PE	000000032f9a0000-000000032f9b7000	Deferred        msimg32
PE	0000000330260000-000000033038f000	Deferred        advapi32
PE	00000003422e0000-0000000342609000	Deferred        urlmon
PE	0000000351680000-000000035168f000	Deferred        api-ms-win-core-localization-l1-2-1
PE	0000000361860000-00000003620cd000	Deferred        windowscodecs
PE	0000000375610000-000000037566c000	Deferred        win32u
PE	00000003a0440000-00000003a0693000	Deferred        wininet
PE	00000003af670000-00000003af9ff000	Deferred        ucrtbase
PE	00000003afd00000-00000003afd68000	Deferred        imm32
PE	00000003be590000-00000003be842000	Deferred        dbghelp
PE	00007f4501cb0000-00007f4501cb4000	Deferred        winex11
Threads:
process  tid      prio (all id:s are in hex)
00000020 start.exe
	00000024    0
00000038 services.exe
	0000003c    0
	00000040    0
	0000004c    0
	0000005c    0
	00000078    0
	00000084    0
	000000b0    0
	000000c8    0
	000000e0    0
00000044 winedevice.exe
	00000048    0
	00000054    0
	00000058    0
	00000060    0
	00000064    0
	000000d4    0
00000068 winedevice.exe
	0000006c    0
	0000007c    0
	00000080    0
	00000088    0
	0000009c    0
	000000a0    0
	000000a4    0
00000070 explorer.exe
	00000074    0
	0000008c    0
	00000090    0
000000a8 plugplay.exe
	000000ac    0
	000000b4    0
	000000b8    0
	000000bc    0
000000c0 svchost.exe
	000000c4    0
	000000cc    0
	000000d0    0
000000d8 rpcss.exe
	000000dc    0
	000000e4    0
	000000e8    0
	000000ec    0
	000000f0    0
	000000f4    0
000000fc (D) C:\Program Files\SumatraPDF\SumatraPDF.exe
	00000100    0 <==
	00000104    0
	00000110    0
System information:
    Wine build: wine-7.0 (Debian 7.0~repack-4)
    Platform: x86_64
    Version: Windows 7
    Host system: Linux
    Host version: 5.18.0-4-amd64

**Comment by GitHubRulesOK** [comment]
Wine is unchartered territory 
and Mobi has many flavor's so a failing sample helps to check for any file format specific related bugs, but since SumatraPDF is  heavily tied to MS GDI Graphics, then for compatibility there may be cross platform issues within Wines graphics sub systems

**Comment by SpearRaven** [comment]
What you say makes sense of course. What I have basically assumed as well.
Thanks.

**Reply by GitHubRulesOK** [reply]
perhaps try the older stable XP 3.1.2 but there were rumblings about tabs i think?

**Comment by SpearRaven** [comment]
Just letting you and other readers know, 3.1.2 had the same issue. I'm
putting this on the backburner for now. Thanks though

**Comment by kjk** [comment]
Unfortunately I have no way to test under Wine and callstack doesn't have symbols so doesn't tell me anything.

**Reply by projectcato** [reply]
Just putting this out there, because I recently did it for something else. Thanks to WSL2 (Windows subsystem for linux), you can run a native linux environment from inside Windows, add WINE to it, and launch full GUI apps.

**Comment by PeterGreb** [comment]
I have no problem with displaying mobi files, SumatraPDFv3.4.6, Wine-7.0.1 but I use the 32-bit version of SumatraPDF.
